src/classlibnative/float/floatdouble.cpp

```diff
@@ -132,6 +137,10 @@ FCIMPL1_V(double, COMDouble::Ceil, double x)
     return (double)ceil(x);
 FCIMPLEND
 
+#if defined(_MSC_VER) && defined(_TARGET_AMD64_)
+#pragma float_control(precise, off)
```

Comment:
In this case, the previous state should already be guaranteed by the pragmas at the top of the file.

I'll update them all to use the push/pop mechanism, however.

Reply:
@AndyAyersMS, there seems to be another bug here 😄 

Attempting to do `float_control(value,setting push)` (for example `#pragma float_control(precise, off push)`), as is documented to work [here](https://docs.microsoft.com/en-us/cpp/preprocessor/float-control?view=vs-2017), results in:
> floatdouble.cpp(42): warning C4083: expected ')'; found identifier 'push'

The best we can do is pushing/popping the entire stack

Reply:
Hmm, too bad. Let the C++ folks know, and then just go with your original version.

Reply:
Working on logging the bug, but I already pushed changes to do:
```
#pragma float_control(push)
#pragma float_control(precise, on)
```
instead of `#pragma float_control(precise, on push)`

Reply:
Actually, it looks to be failing because we have a `#define push` somewhere

Reply:
Logged 673908, expansion of macros in a non-standard pragma is allowed according to the language spec, so it may get closed as "by design".